docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
+architecture, including
 
--  Event-driven - applications base their actions on individual ledger events only.
--  State-driven - applications base their actions on some model of all contracts active on the ledger.
+- DAML React libraries
+- quick iteration against a sandbox backend
+- deployment in the cloud or on `DABL <https://projectdabl.com>`_
+- authentication
 
-Event-driven applications
-=========================
+It provides an excellent starting point to iterate on, so you can build your own application.
 
-**Event-driven** applications react to events on the the ledger and generate commands and other outputs on a per-event basis. They do not require access to ledger state beyond the event they are reacting to.
 
-Examples are sink applications that read the ledger and dump events to an external store (e.g. an external (reporting) database).
+Frontend
+~~~~~~~~
 
-State-driven applications
-=========================
+We recommended building your frontend with the `React <https://reactjs.org>`_ framework. We provide
+several support libraries to build your React frontend for a DAML application.
 
-**State-driven** applications build up a real-time view of the ledger state by reading events and recording contract create and archive events. They then generate commands based on a given state, not just single events.
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
```

Comment:
Can we please add examples how to use the library to the NPM for all mentioned packages? It would really make our documentation and developer experience much nicer than telling the user "go ahead and figure it out on your own". Our testers and internal members really complained when they had to "chace" it elsewhere

Reply:
do you want to have the doc on the npmjs.com packe or here? I thought I don't want to overload this section with details about those libraries here.

Reply:
Not here 😄npmjs.com would be perfect

Reply:
there is actually already a bit of docs there, at least for @daml/react which you should mainly use.

Reply:
https://www.npmjs.com/package/@daml/react

Reply:
@daml/ledger is definitely missing some docs there.

Reply:
I will generate docs for the typescript libraries within the packages and add a section to the building `applications chapter` as a next step, but outside this PR.